File: src/programmers/Lv2/기능개발/solution.js
function solution(progresses, speeds) {
  let max = 0;
  let count = 0;
  let result = [];
  for (let i = 0; i < progresses.length; i++) {
    let leftJob = 100 - progresses[i];
    let leftDay = Math.ceil(leftJob / speeds[i]);
    if (max < leftDay && count != 0) {
      result.push(count);
      count = 1;
    } else {
      count++;
    }
    max = Math.max(max, leftDay);
  }
  result.push(count);

  return result;
}

function solution(progresses, speeds) {
  let answer = [];
  let pro = progresses.slice();
  pro = pro.map((el) => 100 - el);
  pro = pro.map((el, idx) => Math.ceil(el / speeds[idx]));
  let count = 0;
  let num = pro[0];
  for (let i = 0; i < pro.length; i++) {
    if (num >= pro[i]) {
      count++;
      if (i === pro.length - 1) {
        answer.push(count);
      }
    } else {
      num = pro[i];
      answer.push(count);
      count = 1;
      if (i === pro.length - 1) {
        answer.push(count);
      }
    }
  }
  return answer;
}

//console.log(solution([93, 30, 55],[1, 30, 5]));
//console.log(solution([95, 90, 99, 99, 80, 99],[1, 1, 1, 1, 1, 1]));
console.log(solution([96, 94], [3, 3]));

File: src/programmers/Lv2/기능개발/solution.py
import math
class Solution(object):
    @staticmethod
    def solution(progresses, speeds):
        if len(progresses)!=len(speeds):
            return []
        answer = []
        maxNum = 0
        count = 0
        for i in range(len(progresses)):
            rest_num = math.ceil((100-progresses[i])/speeds[i])
            if maxNum>=rest_num:
                count+=1
            else:
                if maxNum == 0:
                    count+=1
                else:
                    answer.append(count)
                    count = 1
            maxNum = max(maxNum,rest_num)
        answer.append(count)
        return answer